src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -322,28 +360,569 @@ TEST_CASE("Datakey and double encryption", "[client_side_encryption]") {
+                                               << concatenate(limits_doc.view()) << finalize;
+
+    // Expect this to fail since encryption results in a document exceeding
+    // the maxBsonObjectSize limit.
+    REQUIRE_THROWS(coll.insert_one(encryption_exceeds_16mib.view()));
 }
 
 TEST_CASE("Views are prohibited", "[client_side_encryption]") {
-    // TODO CXX-2021
+    instance::current();
+
+    // Create a MongoClient without encryption enabled (referred to as client).
+    class client client {
+        uri {}
+    };
+
+    if (test_util::get_max_wire_version(client) < 8) {
+        // Automatic encryption requires wire version 8.
+        WARN("Skipping - max wire version is < 8");
+        return;
+    }
+
+    // Using client, drop and create a view named db.view with an empty pipeline.
+    // E.g. using the command { "create": "view", "viewOn": "coll" }.
+    auto db = client["db"];
+    db["view"].drop();
+
+    auto cmd = make_document(kvp("create", "view"), kvp("viewOn", "coll"));
+    db.run_command(cmd.view());
+
+    // Create a MongoClient configured with auto encryption (referred to as client_encrypted)
+    // Configure with the local KMS provider as follows:
+    // { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+    // Configure with the keyVaultNamespace set to keyvault.datakeys.
+    options::client opts;
+    auto kms_doc = _make_kms_doc();
+    _add_client_encrypted_opts(&opts, {}, std::move(kms_doc));
+    class client client_encrypted {
+        uri{}, std::move(opts)
+    };
+
+    // Using client_encrypted, attempt to insert a document into db.view.
+    // Expect an exception to be thrown containing the message: "cannot auto encrypt a view".
+    REQUIRE_THROWS_MATCHES(client_encrypted["db"]["view"].insert_one(cmd.view()),
+                           mongocxx::exception,
+                           test_util::mongocxx_exception_matcher{"cannot auto encrypt a view"});
 }
 
 TEST_CASE("Corpus", "[client_side_encryption]") {
-    // TODO CXX-2021
+    // TODO CXX-2022
+}
+
+void _round_trip(class client_encryption* client_encryption,
+                 bsoncxx::types::bson_value::view datakey) {
+    auto to_encrypt_doc = document{} << "v"
+                                     << "test" << finalize;
+    auto to_encrypt = to_encrypt_doc.view()["v"].get_value();
+
+    options::encrypt encrypt_opts;
+    encrypt_opts.algorithm(options::encrypt::encryption_algorithm::k_deterministic);
+    encrypt_opts.key_id(datakey.get_binary());
+    auto encrypted_val = client_encryption->encrypt(to_encrypt, encrypt_opts);
+    auto decrypted_val = client_encryption->decrypt(encrypted_val);
+
+    REQUIRE(decrypted_val == to_encrypt);
+}
+
+void _run_endpoint_test(class client_encryption* client_encryption,
+                        bsoncxx::document::view masterkey,
+                        stdx::optional<std::string> error_msg = stdx::nullopt) {
+    options::data_key opts;
+    opts.master_key(masterkey);
```

Comment:
Can you add `INFO (masterkey)`? If this fails, it'll be easier to see which case failed.

Reply:
Done.